### PR TITLE
Add tests for DriveCloudSheet.java

### LINKED ISSUE #32: improve exception handling in CloudDocClient
Use try catch

### PR DESCRIPTION
**Pull request by miasya**

### CODE REVIEW

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  
+  private CloudDocClient cloudDocClient;
+  private CloudSheet cloudSheet;
+
+  @Before
+  public void setUp() {
+    cloudDocClient = new DriveClient();
+    try {
+      cloudSheet = cloudDocClient.getCloudSheet(SPREADSHEET_ID);
+    } catch (IOException e) {
```
Comment (lipczak-g) [inline src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java:76]: I don't understand what is the reason for this catches. In general, your test should be stable enough, so this never happens. Also, setup should not have any assertions - the fail will be communicated with the exception anyway.

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  private static final String RANGE_ROW = "2";
+  private static final String RANGE_COLUMN = "C";
+  //TODO: add invalid range options
+  
+  private CloudDocClient cloudDocClient;
+  private CloudSheet cloudSheet;
+
+  @Before
+  public void setUp() {
+    cloudDocClient = new DriveClient();
```
Comment (lipczak-g) [inline src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java]: Do I get it right that you are establishing full connection to Drive API here? This is a potential source of failures unrelated to the tested code. Typically in a unit test you want to eliminate such dependencies.

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    try {
+      cloudSheet = cloudDocClient.getCloudSheet(SPREADSHEET_ID);
```
Comment (lipczak-g) [inline src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java]: Hmmm... I don't get how are you able to get the cloudSheet object with empty id.

I think instead you should call new DriveCloudSheet(String documentId, Sheets documentService) with the mock of Sheets object. This way you can take DriveClient out of the picture.

**Review by lipczak-g — COMMENTED** [review]
Couple quick comments to unblock you for today.

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java (hunk 1 above)
Comment (miasya) [reply src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java:76]: Updated to handle exceptions in try catch in cloud doc client - so I think that should throw the proper exceptions and cause the fail without the try catch nor assertions in the setup?

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java (hunk 2 above)
Comment (miasya) [reply src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java]: Yes, but I think you're completely right! Will update

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java (hunk 3 above)
Comment (miasya) [reply src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java]: added mocks!

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+  @Test
+  public void testReadRangeSingle() {
+    String[] array = {"B1"};
+    List<List<Object>> values = Arrays.asList(
+      Arrays.asList(array)
+    );
+
+    when(sheetsServiceMock.spreadsheets()).thenReturn(spreadsheetsMock);
+    when(spreadsheetsMock.values()).thenReturn(valuesMock);
+    try {
```
Comment (lipczak-g) [inline src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java:73]: don't understand why the try is needed, but if it is needed please add Assert.fail in the catch block.

File: src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+      when(valuesMock.get(SPREADSHEET_ID, String.format("%s!%s", SHEET_NAME, RANGE_SINGLE))).thenReturn(getMock);
+      when(getMock.execute()).thenReturn(valueRangeMock);
+    } catch (IOException e) {}
+
+    when(valueRangeMock.getValues()).thenReturn(values);
```
Comment (lipczak-g) [inline src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java:80]: super nit: I think you should be able to create a setUpSpreadsheets method where some or all of these could be added, but this is not a hard requirement and some people argue that the setup should be explicit in all test functions even if it creates code bloat.

**Review by lipczak-g — APPROVED** [review]

**Pull request merged**